src/Workspaces/VisualBasic/Portable/Recommendations/VisualBasicRecommendationServiceRunner.vb

```diff
@@ -341,11 +341,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Recommendations
         End Function
 
         ''' <summary>
-        ''' In MemberAccessExpression Contexts, filter out event symbols, except inside AddRemoveHandler Statements
+        ''' In MemberAccessExpression Contexts, filter out event symbols (except for NameOf context), except inside AddRemoveHandler Statements
         ''' Also, filter out any implicitly declared members generated by event declaration or property declaration
         ''' </summary>
-        Private Shared Function FilterEventsAndGeneratedSymbols(node As MemberAccessExpressionSyntax, s As ISymbol) As Boolean
-            If s.Kind = SymbolKind.Event Then
+        Private Function FilterEventsAndGeneratedSymbols(node As MemberAccessExpressionSyntax, s As ISymbol) As Boolean
+            If s.Kind = SymbolKind.Event AndAlso Not _context.IsNameOfContext Then
```

Comment:
I was very specific here to the linked issue and allowed extended events to only NameOf.

Are there any other places where events should show in completion but they currently don't?